aten/src/ATen/Half-inl.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include "ATen/ATenGeneral.h"
+#include <cstring>
+#include <limits>
+
+#ifdef __CUDACC__
+#include <cuda_fp16.h>
+#endif
+
+namespace at {
+
+/// Constructors
+
+inline AT_HOSTDEVICE Half::Half(const __half& value) {
+  std::memcpy(&x, &value, sizeof(unsigned short));
```

Comment:
they have std::memcpy in device code???

Reply:
Yes, although it doesn't seem to be as smart about trivial memcpys so maybe adding a special case for device code makes sense. (On CPU, this ends up being equivalent to `x = value` if you compile with optimizations)